docs/docs/protocol-specs/gas-and-fees/kernel-tracking.md

```diff
@@ -428,6 +429,8 @@ Further, we can trust that the `transaction_fee` the public VM reported is the o
 
 The PublicKernelSetup circuit takes in a `PublicKernelData` and a `PublicCallData` and outputs a `PublicKernelCircuitPublicInputs`.
 
+It must set the `constant_data.global_variables` to the `global_variables` from `PublicCircuitPublicInputs`, as a means to initialize them, since they were zero in the previous kernel coming from private.
```

Comment:
This could occur in app logic or teardown as well right? Can we have it say that it sets the global variables if they did not previously exist and were provided in the `PublicCircuitPublicInputs`, or validates them if they were set previously? Or said differently, it can be set once in any phase, but then the provided value must be present in all subsequent calls.

Reply:
Good catch, updated!